### ISSUE TITLE
False positive conflicting alias error when namespace is the same as the required alias

### ISSUE THREAD

**Issue by rap1ds** [clj-kondo/clj-kondo#2929]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

clj-kondo v2026.07.24

**platform**

Mac

**problem**

I upgraded clj-kondo and started seeing unexpected conflicting alias errors when the namespace name is the same as the alias.

For example, I have a file in `dev/json.clj` which defines a namespace `json` and requires cheshire with alias `json` and this produces a conflicting alias error:

```
(ns json
  "Namespace for dev time JSON utilities, i.e. parsing and pretty printing."
  (:require
   [cheshire.core :as json]))

(defn parse-string
  [s]
  (json/parse-string s))
```

Expected: No conflicting alias errors

Actual: Conflicting alias error

**repro**

```
echo '(ns json (:require [cheshire.core :as json]))' | clj-kondo --config '{:linters {:unused-namespace {:level :off}}}' --lint -
<stdin>:1:39: error: Conflicting alias for cheshire.core
linting took 26ms, errors: 1, warnings: 0
```

**expected behavior**

My expectation is that this shouldn't produce a linting error. AFAIK, the main reason for the conflicting alias error is to catch cases where two namespaces are required with the same alias.

**Comment by borkdude** [comment]
This is a regression, sorry about that. You can try to ignore it with `#_:clj-kondo/ignore`

**Comment by borkdude** [comment]
Hmm, I took another look at this. 
The reason this fires is that it can be confusing when the ns name itself is also used to refer to vars in the namespace.

``` clojure
(ns foo (:require [clojure.set :as foo]))
(def join 3)
foo/join ;; set/join or foo/join?
```

This is why we count the namespace name as an "alias" too in this linter to prevent issues like that.
